### DISCUSSION TITLE
wondering if I can deploy my Vercel WebApp closer to my customer (in EU, US, and Asia)

### DISCUSSION THREAD

**Discussion by pjerryhu** [vercel/community#6608] [Help] [answered]
### Summary

My WebApp speed is slow. In particular, sometimes my First Contentful Paint is slow (>6s).

I'm wondering if I can deploy my Vercel WebApp closer to my customer (in EU, US, and Asia). 
Specifically, I'm wondering 

- How does Vercel WebApp `getServerSideProps` on any given  `web/pages/*` serve content? Do I need to set this to closer to my customer region?

Thanks for answering my question

### Steps to Reproduce

Question about Vercel WebApp under the hood

**Comment by versecafe** [comment, marked as answer]
@pjerryhu ISR would dramatically help for most users assuming the content isn't unique per user, you can also try running pages that don't need to access an external data source on the `edge` runtime to run near users; If you do need to read a DB for some of the data you could try the App router and use PPR to stream in the dynamic content as it's ready, and the static shells would be served on the edge cdn which should keep FCP under 2 seconds globally.

**Reply by pjerryhu** [reply]
Hey @versecafe , thank you so much for the response. I'm learning so much just by asking GPT4 to explain your response lol (not familiar with lots of jargons). 

my `getServerSideProps` for a given web page has 5 interactions with my DB via supabase restAPI. I co-located DB and Vercel serverless functions together, but I suspect that `getServerSideProps` is executed on the client side, which is the cause of long response time. I don't think edge runtime concept applies here. So I'll try out the App router approach you mentioned above

Is there a strategy to optimize LCP (largest contentful paint) too? Appreciate your response

**Reply by versecafe** [reply]
@pjerryhu getServerSideProps should be on the server and if it is running on the client you may first want to make sure you're not leaving open any large security holes from that, LCP best strategy would likely be first the FCP but then for large assets like images you'll want to try using things like `next/image` to optimize them, always set content sizes for them, and then make sure they are coming from a CDN rather than a single point serve, so Vercel Blob or Cloudflare CDN instead of S3 or GCP Bucket, you can also generate blur hashes for images which can load near instantly first and then stream in the image.

**Reply by pjerryhu** [reply]
Thanks so much @versecafe . 

Yea, we use Vercel and NextJS 14, so getServerSideProps is definitely running on server, Vercel just doesn't make it clear on where the server is for us or whether we could move it closer to the customer ISP. 

Our content is mostly text and directly from DB (heavy report content), so CDN strategy is not applicable yet, but we'll think about co-locating read DB replica closer to customer ISP. And also your PPR strategy.
